docs/SmartStore.md

```diff
@@ -0,0 +1,233 @@
+# SmartStore Resource Guide
+
+The Splunk Operator includes support for configuring a SmartStore remote storage volume with index support using a custom resource. The SmartStore integration is not implemented as a StorageClass. This feature and its settings rely on support integrated into Splunk Enterprise. See [SmartStore](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/AboutSmartStore) for information on the feature and implementation considerations.
+
+ * SmartStore support in Kubernetes Operator is limited to Amazon S3 only
+ * You can only define SmartStore-enabled indexes using this specification
+ * You can not convert an existing index into a SmartStore-enabled index using this specification
+ * SmartStore configuration is supported on these Custom Resources: Standalone, and ClusterMaster
+
+
+SmartStore configuration involves indexes, volumes, and the volume credentials. Volume credentials are supplied through a Kubernetes secret object, where as the indexes and volume configurations are provided through the Custom Resource specification. For the CR SmartStore volume spec, the K8 secret object(`secretRef`) is where the volume keys are stored.
+
+
+Here is an example command to encode and load your remote storage volume secret key and access key in the kubernetes secret object: `kubectl create secret generic <secret_store_obj> --from-literal='s3_access_key=<access_key>' --from-literal='s3_secret_key=<secret_key>'`
+  
+
+## Creating a Standalone instance utilizing SmartStore 
+1. Confirm your S3-based storage volume path and URL.
+2. Confirm the name of the Splunk indexes being used with the SmartStore volume. 
+3. Pass the volume and index configuration into the environment through the Standalone Customer Resource specification.
+
+
+Example:
+
+```yaml
+apiVersion: enterprise.splunk.com/v1beta1
+kind: Standalone
+metadata:
+  name: <name>
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 1
+  smartstore:
+    volumes:
+    - name: <remote_volume_name>
+         path: <remote_volume_path>
+         endpoint: https://s3-<region>.amazonaws.com
+         secretRef: <secret_store_obj>
+    indexes:
+      - name: <index_name_1>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_2>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_3>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+```
+
+The SmartStore parameters will be placed into the required .conf files in an app. The app is named as `splunk-operator`. In the case of a standalone deployment, the app is located at `/opt/splunk/etc/apps/`
+ 
+ 
+ 
+## Creating an IndexCluster instance utilizing SmartStore
+1. Create an IndexCluster. See [Examples](Examples.md).
```

Comment:
Is the #1 step needed before Splunk Indexes can be setup? In other words, Admin can always setup the indexes while configuring the IndexerCluster for the first time (via CM) right?

Reply:
Yes, the order is not important here. We mentioned that step, as the steps listed here are about the creation of the indexer cluster.

Reply:
Step 0: Create a Secret object with Secret & Access credentials, as explained in "Storing SmartStore Secrets"
Step 3: Create/Update the Cluster Master Customer Resource specification with volume and index configuration (see Example below)
Step 4: Follow the rest of the steps to Create an Indexer Cluster. See [Examples](Examples.md)